demo/site/src/blocks/YouTubeVideoBlock.tsx

```diff
@@ -27,14 +28,28 @@ const YouTubeVideoBlock: React.FunctionComponent<PropsWithData<YouTubeVideoBlock
         // no url, but ID was specified
     }
 
+    const getSearchParams = () => {
+        const searchParams = new URLSearchParams();
+        searchParams.append("modestbranding", "1");
+
+        searchParams.append("autoplay", Number(autoplay).toString());
+        autoplay && searchParams.append("mute", "1");
+
+        searchParams.append("controls", Number(showControls).toString());
+
+        searchParams.append("loop", Number(loop).toString());
+        loop && searchParams.append("playlist", youtubeIdentifier);
+        return searchParams.toString();
+    };
+
+    const youtubeBaseUrl = "https://www.youtube-nocookie.com/embed/";
+    const youtubeUrl = new URL(`${youtubeBaseUrl}${youtubeIdentifier}`);
+    youtubeUrl.search = getSearchParams();
```

Comment:
I'll wrap it in a useCallback or do you think removing the function all together and use useMemo makes more sense?

Reply:
Personally, I would neither use useMemo or useCallback as long as there's no noticeable need for a performance improvement

Reply:
But does it hurt/ Is there a tradeoff when using useMemo/useCallback? If I just put the code on top level then it will run on every rerender. If it's memoized it only gets run once.

Reply:
It depends 🙂 In this scenario, it increases complexity of the code without any obvious improvements. I agree, running once is an improvement, but the block should only be rendered once anyway (in the best case).

If you're interested, you could have a look at the following resources:

1. useMemo docs: https://react.dev/reference/react/useMemo, especially the deep dive on "How to tell if a calculation is expensive?"
2. Kent C Dodd's blog post on useMemo/useCallback: https://kentcdodds.com/blog/usememo-and-usecallback

However, if you deem useMemo necessary here, it's fine for me 🙂 You could see this as a general advice

Reply:
But in this case, I'd definitely use useMemo over useCallback!

Reply:
Alright 👍🏻  Thank you very much for the resources!